### PR TITLE
Rename describableobject* to accessibleobject*, make roles more flexible

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change is intended to more generally express accessibility aspects of objects as well as to name the scripts / properties more descriptive of their purpose.

The largest change is to replace the ComponentRole enum with a collection of scriptable objects that can be defined by developers to create a custom role / extend the current set.

This change is one in a series that will be used to add immersive screen reader support.

### CODE REVIEW

File: com.microsoft.mrtk.accessibility/Utilities/ScreenReaderSettings.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Accessibility
+{
+    /// <summary>
+    ///
```
Comment (keveleigh) [inline com.microsoft.mrtk.accessibility/Utilities/ScreenReaderSettings.cs:9]: MIssing some summaries

File: com.microsoft.mrtk.accessibility/Utilities/ObjectRoles/UI/Tree.asset [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 0}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: 34c6fa1bae690a3419606a5e4af0f24d, type: 3}
+  m_Name: Tree
+  m_EditorClassIdentifier: 
+  description: ' A type of list that may contain sub-level nested groups that can
```
Comment (keveleigh) [inline com.microsoft.mrtk.accessibility/Utilities/ObjectRoles/UI/Tree.asset:15]: nit: looks like there's a leading space here

File: com.microsoft.mrtk.accessibility/Utilities/ObjectRoles/UI/Application.asset [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 0}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: 34c6fa1bae690a3419606a5e4af0f24d, type: 3}
+  m_Name: Application
+  m_EditorClassIdentifier: 
+  description: ' A structure containing one or more focusable elements requiring
```
Comment (keveleigh) [inline com.microsoft.mrtk.accessibility/Utilities/ObjectRoles/UI/Application.asset:15]: nit: leading whitespace

File: com.microsoft.mrtk.accessibility/Utilities/ObjectRoles/UI/TreeGrid.asset [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 0}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: 34c6fa1bae690a3419606a5e4af0f24d, type: 3}
+  m_Name: TreeGrid
+  m_EditorClassIdentifier: 
+  description: A grid whose rows can be expanded and collapsed in the same manner
```
Comment (keveleigh) [inline com.microsoft.mrtk.accessibility/Utilities/ObjectRoles/UI/TreeGrid.asset:15]: Do we want to keep a link or something to the W3 definitions, like we had before? e.g., https://www.w3.org/TR/wai-aria-1.1/#treegrid

Comment (davidkline-ms) [reply com.microsoft.mrtk.accessibility/Utilities/ObjectRoles/UI/TreeGrid.asset:15]: maybe. i wondered that myself

File: com.microsoft.mrtk.accessibility/Utilities/ScreenReaderSettings.cs (hunk 1 above)
Comment (davidkline-ms) [reply com.microsoft.mrtk.accessibility/Utilities/ScreenReaderSettings.cs:9]: this is a too early file :smile: removing for now

File: com.microsoft.mrtk.accessibility/Utilities/ObjectRoles/UI/TreeGrid.asset (hunk 4 above)
Comment (davidkline-ms) [reply com.microsoft.mrtk.accessibility/Utilities/ObjectRoles/UI/TreeGrid.asset:15]: added.

**Review by keveleigh — APPROVED** [review]

**Pull request merged**